components/camel-stomp/src/main/java/org/apache/camel/component/stomp/StompComponent.java

```diff
@@ -55,6 +67,7 @@ protected Endpoint createEndpoint(String uri, String remaining, Map<String, Obje
         if (config.getSslContextParameters() == null) {
             config.setSslContextParameters(retrieveGlobalSslContextParameters());
         }
+        endpoint.setHeaderFilterStrategy(getHeaderFilterStrategy());
```

Comment:
Ah yeah the jms component has the same problem, you should generally set component options on the endpoint first, and then let endpoint parameters override then if any is provided in the uri. But i guess its rare use-case to use a custom header filter strategy per endpoint and thus not noticed before.